src/actions/replace.py

```diff
@@ -0,0 +1,58 @@
+from talon import Module, actions
+from dataclasses import dataclass
+
+mod = Module()
+
+
+@dataclass
+class RangeGenerator:
+    start: int = 0
+
+    def generate(self, length: int):
+        iter = range(self.start, self.start + length)
+        return [str(i) for i in iter]
+
+
+@mod.capture(rule="count [from <number_small>]")
+def cursorless_count(m) -> str:
+    try:
+        start = m.number_small
+    except:
+        start = 0
+    return RangeGenerator(start)
+
+
+@mod.capture(rule="[<user.formatters>] <user.text> (and <user.text>)*")
+def cursorless_texts(m) -> str:
+    texts = m.text_list
+    try:
+        formatters = m.formatters
+        texts = list(map(
+            lambda text: actions.user.formatted_text(text, formatters),
+            texts
+        ))
+    except:
+        pass
+    return texts
+
+
+@mod.capture(rule="<user.cursorless_texts> | <user.cursorless_count>")
+def cursorless_replace_value(m) -> str:
+    return m[0]
+
+
+@mod.action_class
+class Actions:
+    def cursorless_replace(targets: dict, texts: list[str] or RangeGenerator):
+        """Replaced targets with texts"""
+        if isinstance(texts, RangeGenerator):
+            texts = texts.generate(get_length(targets))
+        actions.user.cursorless_single_target_command(
+            "replace", targets, texts
+        )
+
+
+def get_length(targets: dict):
+    if targets["type"] == "list":
+        return len(targets["elements"])
+    return 1
```

Comment:
This doesn't handle multi-cursor or "every" properly.  Need to either have user specify length or do it extension-side

Reply:
Then we do with extension side. I will get on that

Reply:
Ok yeah curious to see what api you land on